docs/csharp/roslyn-sdk/source-generators-overview.md

```diff
@@ -1,93 +1,75 @@
+Source Generators can improve performance in ways that aren't limited to reflection at run time to discover types as well. Some scenarios involve calling the MSBuild C# task (called CSC) multiple times so they can inspect data from a compilation. As you might imagine, calling the compiler more than once affects the total time it takes to build your app! We're investigating how Source Generators can be used to obviate the need for juggling MSBuild tasks like this, since Source generators don't just offer some performance benefits, but also allows tools to operate at the right level of abstraction.
 
-Another capability Source Generators can offer is obviating the use of some “stringly-typed” APIs, such as how ASP.NET Core routing between controllers and razor pages work. With a Source Generator, routing can be strongly typed with the necessary strings being generated as a compile-time detail. This would reduce the amount of times a mistyped string literal leads to a request not hitting the correct controller.
+Another capability Source Generators can offer is obviating the use of some "stringly-typed" APIs, such as how ASP.NET Core routing between controllers and razor pages work. With a Source Generator, routing can be strongly typed with the necessary strings being generated as a compile-time detail. This would reduce the amount of times a mistyped string literal leads to a request not hitting the correct controller.
 
 ## Get started with source generators
 
 In this guide, you'll explore the creation of a source generator using the <xref:Microsoft.CodeAnalysis.ISourceGenerator> API.
 
-1. Create a .NET console application.
-
-2. Replace the Program class with the following:
+1. Create a .NET console application. This example uses .NET 6.
 
-    ```csharp
-    partial class Program
-    {
-        static void Main(string[] args)
-        {
-            HelloFrom("Generated Code");
-        }
+1. Replace the `Program` class with the following:
 
-        static partial void HelloFrom(string name);
-    }
-    ```
+    :::code language="xml" source="snippets/source-generators/ConsoleApp/Program.cs":::
 
     > [!NOTE]
     > You can run this sample as-is, but nothing will happen yet.
 
-3. Next, we'll create a source generator that will fill in the contents of the `HelloFrom` method.
+1. Next, we'll create a source generator project that will implement the `partial void HelloFrom` method counterpart.
 
-4. Create a .NET standard library project that looks like this:
+1. Create a .NET standard library project that targets the `netstandard2.0` target framework moniker (TFM):
 
-    ```xml
-    <Project Sdk="Microsoft.NET.Sdk">
+    :::code language="xml" source="snippets/source-generators/SourceGenerator/SourceGenerator.csproj":::
```

Comment:
Code language is wrong
```suggestion
    :::code language="csharp" source="snippets/source-generators/SourceGenerator/SourceGenerator.csproj":::
```

Reply:
The _.csproj_ file is XML, this suggestion is incorrect 😄

Reply:
lol